### PR TITLE
Added tag-filter param to support Continuous Delivery

### PR DESCRIPTION
**Pull request by digitalbanana**
I have added a tag-filter parameter to the plugin this will allow the same drone file to be used in a continuous delivery environment where only a single environment (integration in our case) will be published to. Any builds with a tag that does not match the filter will be skipped but not cause the build to fail. Realise this may require tweaking etc but we're using this functionality already so feel its useful.

### CODE REVIEW

**Comment by josmo** [comment]
Thanks for the PR @digitalbanana! I did have a quick question on the use case.  Is there a reason that conditions from drone won't work to select which image to deploy? ie.

```
  deploy:
    image: peloton/drone-rancher
    url: http://serverurl
    secrets: [ rancher_access_key, rancher_secret_key ]
    service: ${DRONE_REPO}
    docker_image: ${DRONE_REPO}:${DRONE_BUILD_NUMBER}
    when:
      branch: whatever realease you're needing
      event: push
```

**Comment by digitalbanana** [comment]
Hey @josmo, yes principally because we're using logic other than the branch info to workout the tag for the image to be published.  So for continuous delivery environments we look at various conditions (branch, commit message etc) to determine the tag of the image to publish to docker and also then whether or not to automatically deploy it. In the same way as the docker plugin works, I altered this fork to look at the contents of the local .tags file to determine if this plugin should upgrade Rancher or not.

Basically built in drone conditions dont offer the flexibility needed to implement this kind of logic

File: Dockerfile [hunk 1]
```diff
@@ -8,9 +8,10 @@ ENV GODEBUG=netdns=go
 COPY --from=alpine /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
 
 LABEL org.label-schema.version=latest
-LABEL org.label-schema.vcs-url="https://github.com/josmo/drone-rancher.git"
+LABEL org.label-schema.vcs-url="https://github.com/croudsupport/drone-rancher.git"
```
Comment (josmo) [inline Dockerfile:11]: should leave this as josmo :)

File: Dockerfile [hunk 2]
```diff
@@ -8,9 +8,10 @@ ENV GODEBUG=netdns=go
...
 LABEL org.label-schema.name="Drone Rancher"
-LABEL org.label-schema.vendor="Josmo"
+LABEL org.label-schema.vendor="Croud"
```
Comment (josmo) [inline Dockerfile:14]: same ... probably doesn't need to be edited at all

File: main.go [hunk 3]
```diff
@@ -2,7 +2,7 @@ package main
 
 import (
 	"fmt"
-	"github.com/urfave/cli"
+	 "github.com/bradrydzewski/cli"
 	"log"
```
Comment (josmo) [inline main.go:6]: probably leave with urfave right now since this is vendored

File: main.go [hunk 4]
```diff
@@ -29,6 +29,8 @@ func main() {
 	app.Action = run
 	app.Version = fmt.Sprintf("1.0.0+%s", build)
 
+	fmt.Println("hello world")
```
Comment (josmo) [inline main.go:32]: probably doesn't need this ;)

File: main.go [hunk 5]
```diff
@@ -103,6 +116,8 @@ func run(c *cli.Context) error {
 		Secret:         c.String("secret-key"),
 		Service:        c.String("service"),
 		DockerImage:    c.String("docker-image"),
+		Tags:    		c.StringSlice("docker-image-tags"),
```
Comment (josmo) [inline main.go:119]: gofmt

File: plugin.go [hunk 6]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/rancher/go-rancher/client"
 	"strings"
 	"time"
+    "regexp"
```
Comment (josmo) [inline plugin.go:10]: gofmt

**Review by josmo — CHANGES_REQUESTED** [review]
Do you mind updating the Docs to give an example on how the logic is supposed to work? :)

File: Dockerfile (hunk 1 above)
Comment (digitalbanana) [reply Dockerfile:11]: yes sure - will do this for the PR

File: Dockerfile (hunk 2 above)
Comment (digitalbanana) [reply Dockerfile:14]: as above

File: main.go (hunk 3 above)
Comment (digitalbanana) [reply main.go:6]: problem here is bradrydzewski has an additonal function to read files from disks so this might be trickier to resolve

File: main.go (hunk 4 above)
Comment (digitalbanana) [reply main.go:32]: ha oops - will remove

File: main.go (hunk 5 above)
Comment (digitalbanana) [reply main.go:119]: will fix

File: plugin.go (hunk 6 above)
Comment (digitalbanana) [reply plugin.go:10]: will fix

**Comment by digitalbanana** [comment]
will update the PR shortly and confirm

**Comment by josmo** [comment]
@digitalbanana Just wanted to let you know that I merged a PR that moved the the V2 apis. So once you get the deps in for the build we'll probably want to pull that in too :)
